postal/service.go

```diff
@@ -39,10 +41,12 @@ type Service struct {
 }
 
 // NewService creates an instance of a Servicel given a Transport.
-func NewService(transport Transport) Service {
+func NewService(transport Transport, platformDir string) Service {
```

Comment:
This changes the public API. Does this mean we want to bump the major version of `packit` as its a backwards-incompatible change?

Reply:
Yes, that's correct. Do we have a way of recording this? (Could be a good case for something like https://github.com/semantic-release/semantic-release eventually)

Reply:
I guess I bring this up because we *just* cut a `1.0`. I feel like we should be doing a better job of keeping some stability in the API for a while before we break things again. Is there a way we can make this backwards-compatible?

Reply:
Ah, shoot. I see. Let me look and see if there's a way. The `MappingResolver` interface has changed too, so technically `WithDependencyMappingResolver` isn't backwards compatible either...

Reply:
I wrote all this assuming a major version bump would result, but I didn't realize we just cut a `1.0` :(

Reply:
The `MappingResolver` is in an internal package and therefore not part of the public API. I wouldn't worry about that change.

Reply:
True that our mapping resolver implementation is internal, but `WithDependencyMappingResolver` and the `MappingResolver` interface are still public. That said, it would only break folks who are using their own implementation of a mapping resolver, as opposed to ours, so this probably isn't a big deal. In fact, given that, I'm not even sure any library consumer would ever use `WithDependencyMappingResolver` in the first place. I'm guessing we added for DI purposes for our own tests.

Reply:
Here are our options for making this backwards-compatible:
- Hardcode the platform dir here so that we don't need to pass it in. That said, this doesn't actually seem feasible as it is passed in as an argument to build/detect.
- Since the platform dir needs to be fed in somewhere, it seems the only other option is to move it to a parameter on `Resolve`/`ResolveOne`. I could update the lazy-loading logic to store the last used platform dir and only re-read bindings from the filesystem if it has changed.

There may be some other more clever options if you have any other suggestions @ryanmoran

Reply:
@ryanmoran Created a new commit with the second option implemented. Let me know your thoughts. Should be easy to revert/adjust since I left it separate.